sdk/monitor/monitor-opentelemetry/src/metrics/quickpulse/utils.ts

```diff
@@ -219,80 +253,255 @@ function getIso8601Duration(milliseconds: number): string {
+    if (httpStatusCode) {
+      requestData.ResponseCode = Number(httpStatusCode);
     }
+  } else if (grpcStatusCode) {
+    requestData.ResponseCode = Number(grpcStatusCode);
+  }
+  requestData.Success = span.status.code !== SpanStatusCode.ERROR && requestData.ResponseCode < 400;
+  return requestData;
+}
 
-    document = {
-      documentType: KnownDocumentType.Request,
-      name: span.name,
-      url: url,
-      responseCode: code,
-      duration: getIso8601Duration(hrTimeToMilliseconds(span.duration)),
-    };
-  } else {
-    url = getUrl(span.attributes);
+// A slightly modified version of createDependencyData from spanUtils in exporter
+function getDependencyData(span: ReadableSpan): DependencyData {
+  const dependencyData: DependencyData = {
+    Target: "",
+    Duration: hrTimeToMilliseconds(span.duration),
+    Success: span.status.code !== SpanStatusCode.ERROR,
+    Name: span.name,
+    ResultCode: 0,
+    Type: "",
+    Data: "",
+    CustomDimensions: createCustomDimsFromAttributes(span.attributes),
+  };
+
+  if (span.kind === SpanKind.PRODUCER) {
+    dependencyData.Type = DependencyTypes.QueueMessage;
+  }
+  if (span.kind === SpanKind.INTERNAL && span.parentSpanId) {
+    dependencyData.Type = DependencyTypes.InProc;
+  }
+
+  const httpMethod = span.attributes[SEMATTRS_HTTP_METHOD];
+  const dbSystem = span.attributes[SEMATTRS_DB_SYSTEM];
+  const rpcSystem = span.attributes[SEMATTRS_RPC_SYSTEM];
+  // HTTP Dependency
+  if (httpMethod) {
+    const httpUrl = span.attributes[SEMATTRS_HTTP_URL];
+    if (httpUrl) {
+      try {
+        const dependencyUrl = new URL(String(httpUrl));
+        dependencyData.Name = `${httpMethod} ${dependencyUrl.pathname}`;
+      } catch (ex: any) {
+        /* no-op */
+      }
+    }
+    dependencyData.Type = DependencyTypes.Http;
+    dependencyData.Data = getUrl(span.attributes);
     const httpStatusCode = span.attributes[SEMATTRS_HTTP_STATUS_CODE];
     if (httpStatusCode) {
-      code = String(httpStatusCode);
+      dependencyData.ResultCode = Number(httpStatusCode);
+    }
+    let target = getDependencyTarget(span.attributes);
+    if (target) {
+      try {
+        // Remove default port
+        const portRegex = new RegExp(/(https?)(:\/\/.*)(:\d+)(\S*)/);
+        const res = portRegex.exec(target);
+        if (res !== null) {
+          const protocol = res[1];
+          const port = res[3];
+          if (
+            (protocol === "https" && port === ":443") ||
+            (protocol === "http" && port === ":80")
+          ) {
+            // Drop port
+            target = res[1] + res[2] + res[4];
+          }
+        }
+      } catch (ex: any) {
+        /* no-op */
+      }
+      dependencyData.Target = `${target}`;
+    }
+  }
+  // DB Dependency
+  else if (dbSystem) {
+    // TODO: Remove special logic when Azure UX supports OpenTelemetry dbSystem
+    if (String(dbSystem) === DBSYSTEMVALUES_MYSQL) {
+      dependencyData.Type = "mysql";
```

Comment:
It would be good to have an enum here for dependency types, including not DB ones